File: typescript-connector/src/models/QuantitativeValue.ts
// Class from DFC Business Ontology: #QuantitativeValue
import { SemanticObject } from "../core/SemanticObject.js";
import { RepresentedThing, type RepresentedThingParams } from "./RepresentedThing.js";

export interface QuantitativeValueParams extends RepresentedThingParams {
  value?: number;
  hasUnit?: string;
}
export class QuantitativeValue extends RepresentedThing {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:QuantitativeValue";
  }

  value?: number;
  hasUnit?: string;

  constructor(
    semanticId: string,
    params?: QuantitativeValueParams,
  ) {
    super(semanticId, params);
    this.value = params?.value;
    this.hasUnit = params?.hasUnit;
    this.semanticType = QuantitativeValue.SEMANTIC_TYPE;
    this.registerSemanticProperty("dfc-b:QuantitativeValue:value", () => this.value);
    this.registerSemanticProperty("dfc-b:QuantitativeValue:has_unit", () => this.hasUnit);
  }
  static {
    SemanticObject.typeRegistry.set(QuantitativeValue.SEMANTIC_TYPE, QuantitativeValue);
  }
}

File: ruby-gem/lib/models/quantitative_value.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #QuantitativeValue
require_relative '../core/semantic_object'
require_relative 'represented_thing'


module DfcLinkmlConnector
  module Models
    class QuantitativeValue < RepresentedThing
      SEMANTIC_TYPE = "dfc-b:QuantitativeValue".freeze

      # @return [Float]
      attr_accessor :value

      # @return [String]
      attr_accessor :unit

      # @param semanticId [String]
      # @param date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, value: nil, unit: nil
      def initialize(semanticId, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, value: nil, unit: nil)
        super(semanticId, date: date, description: description, name: name, characteristicOf: characteristicOf, dimension: dimension)
        @value = value
        @unit = unit
        self.semanticType = "dfc-b:QuantitativeValue"
        registerSemanticProperty("dfc-b:value", &method("value")).valueSetter = method("value=")
        registerSemanticProperty("dfc-b:hasUnit", &method("unit")).valueSetter = method("unit=")
      end

      Core::SemanticObject.type_registry[SEMANTIC_TYPE] = self
    end
  end
end
